File: Fall Week 3 Greedy/python/solutions/D.py
def solve():
    # read the input
    n = int(input())

    # Denominations
    bills = [100, 20, 10, 5, 1]
    count = 0

    for bill in bills:
        count += n // bill  # Add the number of bills of this denomination
        n %= bill           # Update remaining amount

    print(count)

# Main function to handle a single test case
solve()

File: Fall Week 3 Greedy/solutions/D.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    // read the input
    int n;
    cin >> n;

    // Denominations
    vector<int> bills = {100, 20, 10, 5, 1};
    int count = 0;

    for (int bill : bills) {
        count += n / bill;  // Add the number of bills of this denomination
        n %= bill;          // Update remaining amount
    }

    cout << count << endl;
}

int main() {
    solve();
}
